### PR TITLE
Feat(passage): Passage authoring initial UI PD-3770

### PR DESCRIPTION
**Pull request by dianagociu**
https://illuminate.atlassian.net/browse/PD-3770

### CODE REVIEW

File: packages/passage/configure/src/design.jsx [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+    const updatedModel = { ...model, passages: updatedPassages };
+
+    onModelChanged(updatedModel);
+  };
+
+  changeText = (text, index = 0) => {
+    const { model, onModelChanged } = this.props;
+
+    const updatedPassages = model.passages.map((passage, i) => {
```
Comment (andreeapescar) [inline packages/passage/configure/src/design.jsx:89]: I guess we don't need map, we only set at [index].
Also, sanity checks

File: packages/passage/configure/src/design.jsx [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+      passages = [{
+        title,
+        subtitle,
+        text,
+        author,
+        teacherInstructions,
+      }]
+    } = model || {};
+    const { teacherInstructions: teacherInstructionsError, title: titleError, subtitle: subtitleError, author: authorError, text: textError } = errors;
+
```
Comment (andreeapescar) [inline packages/passage/configure/src/design.jsx:134]: errors || {}

File: packages/passage/controller/src/index.js [hunk 3]
```diff
@@ -1,5 +1,21 @@
 import cloneDeep from 'lodash/cloneDeep';
+import defaults from './defaults';
 
+const getContent = (html) => (html || '').replace(/(<(?!img|iframe)([^>]+)>)/gi, '');
+
+export function createDefaultModel(model = {}) {
```
Comment (andreeapescar) [inline packages/passage/controller/src/index.js:6]: This doesn't seem to be needed?

File: packages/passage/controller/src/index.js [hunk 4]
```diff
@@ -1,5 +1,21 @@
...
+  return new Promise((resolve) => {
+    resolve({
+      ...defaults,
+      ...model,
+    });
+  });
+}
+
+export const normalize = (question) => ({
```
Comment (andreeapescar) [inline packages/passage/controller/src/index.js:15]: Let's make we use normalize

File: packages/passage/docs/pie-schema.json.md [hunk 5]
```diff
@@ -12,10 +12,54 @@ The array object has the following properties:
...
+
+# `textEnabled` (boolean, required)
+
+Indicates if text is enabled
+
+# `authorEnabled` (boolean, required)
+
+Indicates if author is enabled
+
+# `spanishInput` (boolean)
```
Comment (andreeapescar) [inline packages/passage/docs/pie-schema.json.md:51]: Do we need this?

File: packages/passage/docs/pie-schema.json.md [hunk 6]
```diff
@@ -12,10 +12,54 @@ The array object has the following properties:
...
+
+Whether a control to allow insertion of spanish characters should be displayed
+
+Default: `false`
+
+# `specialInput` (boolean)
```
Comment (andreeapescar) [inline packages/passage/docs/pie-schema.json.md:57]: And this

**Review by andreeapescar — CHANGES_REQUESTED** [review]

**Comment by andreeapescar** [comment]
@dianagociu can you please add:
1. screenshot/video rec with authoring
2. [optional]: screenshot with how documentation looks like

**Comment by dianagociu** [comment]
1. Passage Authoring Design
Screenshot 2024-06-24 at 11 37 50
1.1Passage Authoring Settings
Screenshot 2024-06-24 at 11 38 02
2.1 Model for the @pie-elements/passage
Screenshot 2024-06-24 at 12 15 29
Screenshot 2024-06-24 at 12 15 33
Config Object for @pie-elements/passage-configure
Screenshot 2024-06-24 at 12 15 39

**Pull request merged**